plugins/lyrics-genius/back.js

```diff
@@ -17,17 +21,52 @@ module.exports = async (win) => {
 	});
 };
 
+const toggleRomanized = () => {
+	revRomanized = !revRomanized;
+	console.log("Romanized Mode: " + revRomanized);
+};
+
 const fetchFromGenius = async (metadata) => {
-	const queryString = `${cleanupName(metadata.artist)} ${cleanupName(
-		metadata.title
-	)}`;
-	let response = await fetch(
-		`https://genius.com/api/search/multi?per_page=5&q=${encodeURI(queryString)}`
-	);
+	const songTitle = `${cleanupName(metadata.title)}`;
+	const songArtist = `${cleanupName(metadata.artist)}`;
+
+	/* Tried using regex to test the title and artist for East Asian Characters. It works but I realized 
+	some groups are fully English in both title and singer. In this case, the lyrics are  Might improve this in the future.
+	*/
+	let regexEastAsianChars = new RegExp("[\u{3040}-\u{30ff}\u{3400}-\u{4dbf}\u{4e00}-\u{9fff}\u{f900}-\u{faff}\u{ff66}-\u{ff9f}]");
+	let hasAsianChars = false;
+	if(revRomanized) {
+		hasAsianChars = regexEastAsianChars.test(songTitle) || regexEastAsianChars.test(songArtist);
+	}
+
+	let queryString = revRomanized && hasAsianChars? 
+	`${songArtist} ${songTitle} Romanized` : 
+	`${songArtist} ${songTitle}`;
+	console.log(queryString);
+
+	let url = await getSongs(queryString);
+	if (is.dev()) {
+		console.log("Fetching lyrics from Genius:", url);
+	}
+	let lyrics = await getLyrics(url);
+
+	if(revRomanized && !hasAsianChars && regexEastAsianChars.test(lyrics)) {
+		queryString = `${songArtist} ${songTitle} Romanized`;
+		let url = await getSongs(queryString);
+		lyrics = await getLyrics(url);
+	}
+	return lyrics;
```

Comment:
The same code is repeated twice here. You really need to cleanup this whole `fetchFromGenius` function.
Think about making it as short as possible.

Reply:
Got it